### PR TITLE
Ability to not merge into DataContext entities loaded by certain data loaders

### PR DESCRIPTION
**Pull request by knstvk**
See #2250 

The change is trivial - added `readOnly` attribute to data loaders XML. If it's set to true, the loader doesn't get a reference to DataContext and doesn't merge entities after load.

Consequences:
- We will generate `readOnly="true"` for collection loaders by default. In most cases collection containers contain read-only data (list views, option containers). Project and view templates are updated in this PR. We will create an issue for Studio to support this attribute in data container wizards.
- We don't need `<data readOnly="true">` and `NoopDataContext`, instead we'll use `readOnly="true"` for loaders. It is more clear and less error-prone. There will be one `DataContext` implementation in use, it saves everything that is merged into it. Perhaps we'll deprecate `NoopDataContext` in the future and remove `readOnly` attribute from `data` element.

See examples in https://github.com/jmix-projects/master-detail-views-prototype project.

### CODE REVIEW

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-templates/content/project/application/src/main/resources/${project_rootPath}/view/user/user-list-view.xml [hunk 1]
```diff
@@ -2,11 +2,11 @@
 <view xmlns="http://jmix.io/schema/flowui/view"
       title="msg://UserListView.title"
       focusComponent="usersDataGrid">
-    <data readOnly="true">
+    <data>
```
Comment (glebfox) [inline jmix-templates/content/project/application/src/main/resources/${project_rootPath}/view/user/user-list-view.xml:5]: The same change is needed for the Kotlin project template

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/meta/element/StudioDataElements.java [hunk 2]
```diff
@@ -69,6 +69,7 @@ public interface StudioDataElements {
                             defaultValue = "0"),
                     @StudioProperty(xmlAttribute = "maxResults", type = StudioPropertyType.INTEGER,
                             defaultValue = "0"),
+                    @StudioProperty(xmlAttribute = "readOnly", type = StudioPropertyType.BOOLEAN),
```
[the remark this answers is not in the document]
Comment (knstvk) [reply jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/meta/element/StudioDataElements.java]: Done

File: jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/meta/element/StudioDataElements.java [hunk 3]
```diff
@@ -90,6 +91,7 @@ public interface StudioDataElements {
                             defaultValue = "0"),
                     @StudioProperty(xmlAttribute = "cacheable", type = StudioPropertyType.BOOLEAN,
                             defaultValue = "false"),
+                    @StudioProperty(xmlAttribute = "readOnly", type = StudioPropertyType.BOOLEAN),
```
[the remark this answers is not in the document]
Comment (knstvk) [reply jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/meta/element/StudioDataElements.java]: Done

File: jmix-templates/content/project/application/src/main/resources/${project_rootPath}/view/user/user-list-view.xml (hunk 1 above)
Comment (knstvk) [reply jmix-templates/content/project/application/src/main/resources/${project_rootPath}/view/user/user-list-view.xml:5]: Done

**Review by glebfox — APPROVED** [review]

**Comment by knstvk** [comment]
Merged in IDE.

**Pull request closed without merging**